source/module_io/read_input_item_elec_stru.cpp

```diff
@@ -453,47 +453,35 @@ void ReadInput::item_elec_stru()
                           "set to 1, a fast algorithm is used";
         read_sync_bool(input.gamma_only);
         item.reset_value = [](const Input_Item& item, Parameter& para) {
-            Input_para& input = para.input;
-            std::string& basis_type = input.basis_type;
-            bool& gamma_only = input.gamma_only;
-            if (basis_type == "pw" && gamma_only) // pengfei Li add 2015-1-31
+            if (para.input.gamma_only && para.input.basis_type == "pw")
+            {
+                para.input.gamma_only = false;
+               GlobalV::ofs_warning << "ReadInput pw not supported in gamma_only mode."<< std::endl;
+            }
+        };
+        item.check_value = [](const Input_Item& item, const Parameter& para) {
+            if (para.inp.gamma_only && para.inp.basis_type == "pw") 
             {
-                gamma_only = false;
                 GlobalV::ofs_warning << " WARNING : gamma_only has not been "
                                         "implemented for pw yet"
-                                     << std::endl;
+                                        << std::endl;
                 GlobalV::ofs_warning << " the INPUT parameter gamma_only has been reset to 0" << std::endl;
                 GlobalV::ofs_warning << " and a new KPT is generated with "
                                         "gamma point as the only k point"
-                                     << std::endl;
+                                        << std::endl;
 
-                GlobalV::ofs_warning << " Auto generating k-points file: " << input.kpoint_file << std::endl;
-                std::ofstream ofs(input.kpoint_file.c_str());
+                GlobalV::ofs_warning << " Auto generating k-points file: " << para.inp.kpoint_file << std::endl;
+                std::ofstream ofs(para.inp.kpoint_file.c_str());
                 ofs << "K_POINTS" << std::endl;
                 ofs << "0" << std::endl;
                 ofs << "Gamma" << std::endl;
                 ofs << "1 1 1 0 0 0" << std::endl;
                 ofs.close();
             }
-            else if (basis_type == "lcao" && gamma_only == 1)
-            {
-                para.sys.gamma_only_local = true;
-                // std::cout << "gamma_only_local =" << gamma_only_local <<
-                // std::endl;
-                if (input.esolver_type == "tddft")
-                {
-                    GlobalV::ofs_running << " WARNING : gamma_only is not applicable for tddft" << std::endl;
-                    para.sys.gamma_only_local = false;
-                }
-            }
-
-            if ((input.out_mat_r || input.out_mat_hs2 || input.out_mat_t || input.out_mat_dh || input.out_hr_npz
-                 || input.out_dm_npz || input.dm_to_rho)
-                && para.sys.gamma_only_local)
+            if (para.input.nspin ==4 && para.sys.gamma_only_local)
```

Comment:
move it to nspin

Reply:
Now,the cheak_value has been setted in the npsin